src/Transport.ts

```diff
@@ -701,3 +715,17 @@ export function lowerCaseHeaders (oldHeaders?: http.IncomingHttpHeaders): http.I
   }
   return newHeaders
 }
+
+/**
+ * Function for calculating how long to sleep, in seconds, before the next request retry
+ * Uses the AWS "equal jitter" algorithm noted in this post:
+ * https://aws.amazon.com/blogs/architecture/exponential-backoff-and-jitter/
+ * @param min The minimum number of seconds to wait
+ * @param max The maximum number of seconds to wait
+ * @param attempt How many retry attempts have been made
+ * @returns The number of seconds to wait before the next retry
+ */
+function retryBackoff (min: number, max: number, attempt: number): number {
+  const ceiling = Math.min(max, 2 ** attempt) / 2
+  return ceiling + ((Math.random() * (ceiling - min)) - min)
```

Comment:
The expression `((Math.random() * (ceiling - min)) - min)` generates a random number between `-min` and `ceiling - 2 * min`. Was this what you intended here?

Reply:
Nice catch. :laughing: Fixed in 003353e16566db51905abbc8d6b2355361d8d17e.